src/messaging/interface/verifier/response.ts

```diff
@@ -0,0 +1,73 @@
+import { Credential, Did } from '@kiltprotocol/sdk-js'
+import { DidResolveKey, ICredential } from '@kiltprotocol/types'
+
+import { ISession, IEncryptedMessage, ISubmitCredential } from '../../../types'
+import { decrypt, encrypt, assertKnownMessage, fromBody } from '../../index'
+import { isIRequestCredential } from '../../../utils'
+
+/**
+ * Submits credentials as a response to a request credential message.
+ * @param credentials - An array of credentials to be submitted.
+ * @param encryptedMessage - The encrypted message received as part of the message workflow.
+ * @param session - An object containing session information.
+ * @param session.decryptCallback - A callback function used for decryption.
+ * @param session.senderEncryptionKeyUri - The URI of the sender's encryption key.
+ * @param session.receiverEncryptionKeyUri - The URI of the receiver's encryption key.
+ * @param session.encryptCallback - A callback function used for encryption.
+ * @param session.signCallback - A callback function used for signing.
+ * @param options - Additional options for the function.
+ * @param options.resolveKey - A function for resolving keys. (Optional) Used for tests only.
+ * @throws Error if the decrypted message is not a request credential message.
+ * @throws Error if credentials do not match.
+ * @returns A promise that resolves to an encrypted message containing the submitted credentials.
+ */
+export async function submitCredential(
+  credentials: ICredential[],
+  encryptedMessage: IEncryptedMessage,
+  { decryptCallback, senderEncryptionKeyUri, receiverEncryptionKeyUri, encryptCallback, signCallback }: ISession,
+  {
+    resolveKey = Did.resolveKey,
+  }: {
+    resolveKey?: DidResolveKey
+  } = {}
+): Promise<IEncryptedMessage<ISubmitCredential>> {
+  const decryptedMessage = await decrypt(encryptedMessage, decryptCallback, { resolveKey })
+  assertKnownMessage(decryptedMessage)
+
+  if (!isIRequestCredential(decryptedMessage)) {
+    throw new Error('Wrong message. Expected request credential message')
+  }
+
+  const { challenge, cTypes, owner } = decryptedMessage.body.content
+
+  const content = await Promise.all(
+    cTypes.map(async (ctype) => {
```

Comment:
Can we verify that these are valid ctypes?

Reply:
Only the hash is present in the ctype object. I validate the hash's existence by querying it in : [a9ece08](https://github.com/KILTprotocol/kilt-extension-api/pull/24/commits/a9ece0891b7b4e965ae0a743ea3e89f4e4056e9d)